### PR TITLE
Quality improvements for Anya reconstruction

### PR DESCRIPTION
**Pull request by claforte**
See https://github.com/ashawkey/stable-dreamfusion/issues/263 (2023-05-12_01) for detailed results

- add run_image_anya.sh
  -  a first attempt at reconstructing a high-quality Anya-like character
- implement save_guidance for zero123 (thanks Vikram!)
- support save_guidance for batch_size>1
- configurable jitter. 
  - default behavior is preserved for now, but I recommend we consider reducing the default (see Anya results for details)
- configurable grad_scale to 1.0 in zero123_utils.py.
  - previous behavior is preserved, but I don't think that adjusting the grad_scale based on angles is a good idea. I think it might prevent front-facing 3D structure from emerge.
- pass `opt` to some lower-level classes, instead of passing yet more arguments that just come from `opt` anyway

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -61,6 +61,9 @@ def __call__ (self, parser, namespace, values, option_string = None):
     parser.add_argument('--latent_iter_ratio', type=float, default=0.2, help="training iters that only use albedo shading")
     parser.add_argument('--albedo_iter_ratio', type=float, default=0, help="training iters that only use albedo shading")
     parser.add_argument('--jitter_pose', action='store_true', help="add jitters to the randomly sampled camera poses")
+    parser.add_argument('--jitter_center', type=float, default=0.2, help="amount of jitter to add to sampled camera pose's center (camera location)")
```
Comment (voletiv) [inline main.py:64]: Let's make the default 0.015? Also, set the defaults for the rest of them

Comment (claforte) [reply main.py:64]: Let's see if @ashawkey knows of a counter-argument...

Comment (ashawkey) [reply main.py:64]: I think these hyper-parameters can only be decided by running some experiments, do you have any results that shows 0.2 work well?

**Comment by ashawkey** [comment]
The view-angle based grad_scale thing is quite empirical, so a configurable option is convenient for ablation.
I have no problems and you can merge it directly. Thanks!

File: main.py (hunk 1 above)
Comment (claforte) [reply main.py:64]: 0.2 was your default. ;-) In the Anya case study I included screenshots with different jitters: https://github.com/ashawkey/stable-dreamfusion/issues/263

Comment (claforte) [reply main.py:64]: There's no rush to choose a different jitter default, we can do it later...

**Pull request merged**